### DISCUSSION TITLE
RFC: Moving Connectors to a Separate Package, extracting connectors from `@cosmos-kit/react`

### DISCUSSION THREAD

**Discussion by dalechyn** [graz-sh/graz#122] [Ideas]
This RFC (Request for Comments) document outlines the updates to the current architecture of adapters (aka connectors) to minimize the package bloating, yet enabling other projects to use all adapters (aka connectors) in a single package to implement a better wallet connection experience.

Also, since the focus of this RFC is connectors, it is proposed to implement all the connectors that exist in `@cosmos-kit/react` as such proposes a variety of connectors, while `graz` implements only some of them.

## Motivation
Given that `wagmi` efficiently manages its connectors by separating all of them under [`@wagmi/connectors`](https://github.com/wagmi-dev/wagmi/tree/main/packages/connectors) package, it seems beneficial for `graz` to adopt a similar approach. This would reduce amount of packages, yet would still tree-shake effectively since `graz` would not introduce `index.ts` barrel files for such, reducing the consumer's bundle size, and keeping the `graz` as a codebase more granular.

As well as adding more connectors would inspire more FE engineers to use the library since it would offer the same variety of connectors that `@cosmos-kit/react` has, yet offering a better developer experience using the variety of hooks `graz` offers.

## Goals
The primary objectives are:
- Create a `@graz-sh/connectors` package which would have all the existing adapters moved into;
- Add more connectors that `@cosmos-kit/react` offers to have a better variety of wallets to connect to;
- Renaming the `Adapter` to `Connector`, as well as getting rid of unnecessary interface extension since `Connector` is not being used anywhere, thus being an useless export, and getting rid of `Graz` prefix as it is implicitly known for a consumer that such a type comes from `graz`:
  https://github.com/graz-sh/graz/blob/137b2dce915619e6fb44ed1e0fd4d4f20146b6f8/packages/graz/src/adapter/index.ts#L11-L23

**Comment by codingki** [comment]
I like this idea, Are you going to seperate every connector: `connectors/keplr` or just `/connectors`?

**Reply by dalechyn** [reply]
For the first pass I think just `connectors` would be sufficient, though later on we could implement a similar approach `wagmi` has with generating proxy packages.
